src/serializers/type_serializers/with_default.rs

```diff
@@ -74,4 +80,16 @@ impl TypeSerializer for WithDefaultSerializer {
     fn get_default(&self, py: Python) -> PyResult<Option<PyObject>> {
         self.default.default_value(py)
     }
+
+    fn compare_with_default(&self, py: Python, value: &PyAny) -> PyResult<bool> {
+        if let Some(default) = self.get_default(py)? {
+            if let Some(default_comparison) = &self.default_comparison {
+                return default_comparison.call(py, (value, default), None)?.extract::<bool>(py);
```

Comment:
I'm passing the values as positional here, not sure if is the best way to call the comparison operator